pkg/vsphere/vm/vm.go

```diff
@@ -763,11 +763,32 @@ func (vm *VirtualMachine) CheckHostCompatibility(ctx context.Context, host *type
 // to the most suitable host in the cluster. If relocation or subsequent power-on fail, it will attempt the next
 // best host, and repeat this process until a successful power-on is achieved or there are no more hosts to try.
 func (vm *VirtualMachine) PowerOn(op trace.Operation) error {
+	// if we aren't in VC, we don't need to recommend or use DRS-aware power-on
+	if !vm.IsVC() {
+		_, err := vm.WaitForResult(op, func(op context.Context) (tasks.Task, error) {
+			return vm.VirtualMachine.PowerOn(op)
+		})
+		return err
+	}
+
 	drs := vm.Session.DRSEnabled != nil && *vm.Session.DRSEnabled
-	cls := vm.InCluster(op)
+
+	h, err := vm.Cluster.Hosts(op)
+	if err != nil {
+		return err
+	}
+
+	// we only recommend if the VM is in a cluster with more than one host
+	cls := vm.InCluster(op) && len(h) > 1
 
 	op.Debugf("DRS enabled: %t", drs)
-	op.Debugf("%s resides in a cluster: %t", vm.Reference().String(), cls)
+	op.Debugf("%s resides in a multi-host cluster: %t", vm.Reference().String(), cls)
+
+	// if DRS is enabled or we aren't in a multi-host cluster, there is no
+	// reason to recommend a host, so bail early and power-on.
+	if drs || !cls {
```

Comment:
Is this the only place where these are evaluated?  If so I lean toward having the actual comparisons here for readability...if you keep them I'd at least recommend relocating them just above the if with comments...